lib/pluggy_elixir/auth.ex

```diff
@@ -0,0 +1,51 @@
+defmodule PluggyElixir.Auth do
+  @moduledoc """
+  Handle authentication actions.
+  """
+
+  alias PluggyElixir.Config
+  alias PluggyElixir.Error
+
+  @type t :: %__MODULE__{
+          api_key: binary()
+        }
+
+  defstruct [:api_key]
+
+  @auth_path "/auth"
+
+  @spec create_api_key :: {:ok, t()} | {:error, binary()}
```

Comment:
Same as the thread https://github.com/brainnco/pluggy_elixir/pull/1/files#r620286930

I think we should put `@spec` after `doc`, closer to the function itself.

Reply:
👍🏻